### ISSUE TITLE
Dont show false positive for builitin mjml class mj-carousel-main

### ISSUE THREAD

**Issue by timo-reymann** [timo-reymann/intellij-mjml-support#15]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I use a group of MJML-provided classes to style the `<mj-carousel>` component. One is `.mj-carousel-main`. I get feedback from the editor, "Selector mj-carousel-main is never used".

Perhaps that report is unneeded? It's correct in that I don't define the class. Perhaps it's unneeded because MJML defines the class and I see no problem with the condition I described.

### Code

```xml
<mj-style inline="inline" >
.mj-carousel-main { 
    padding-bottom: 20px 
}
</mj-style>
```

### Background
I'm under-fond of some aspects of MJML styling for the <mj-carousel> component. MJML includes this class as part of the HTML code produced for the component. The class is not documented; it is easy to find in the Debugger Tool. It is a useful starting point for targeting CSS to parts of the DOM at which I can create my preferred styling.

 
#### I'd like
To be able to use this technique without seeing a Warning.

#### I see instead
A Warning, _Selector mj-carousel-main is never used_.

#### Discussion
The report is strictly correct: I do not include the target of this class in my code. The report is improvable: The report doesn't identify either a troubling condition or a problem worthy of a solution.

> Reported by Garry via Mail

**Comment by GarryFlemings** [comment]
I use these four MJML-defined classes for personal styling:

```
mj-carousel-main (mentioned above)
mj-carousel-previous
mj-carousel-next
mj-carousel-thumbnail
```

I suggest putting them in scope for this possible change.

I observe MJML also defines the classes below. I have not assessed whether these should be in scope.

```
mj-carousel-radio
mj-carousel-content
mj-carousel-previous-icons
mj-carousel-images
mj-carousel-image
mj-carousel-next-icons
```

**Comment by timo-reymann** [comment]
Maintainance should be very easy and the risk of "false usage detection" should be relatively low, but gives the user a better feedback and better integration with mjml.

Solution to support this in a generic manner and also for custom tags:

1. Extend model with used classes
2. Add CssClassOrIdReferenceBasedUsagesProvider like  [MjmlCssClassUsageProvider](https://github.com/timo-reymann/intellij-mjml-support/blob/main/src/main/kotlin/de/timo_reymann/mjml_support/reference/MjmlCssClassUsageProvider.kt) that contributes predefined classes automatically
3. Add for builtin tags 

When it works with the custom CssClassOrIdReferenceBasedUsagesProvider should be very easy as above, if this doesnt work further research is needed.

**Comment by timo-reymann** [comment]
Its pretty simple with the CssClassOrIdReferenceBasedUsagesProvider as stated in my previous comment.

I added the classes you mentioned for the mj-carousel component @GarryFlemings.

Do you know if there are additional cases/components where this would make sense?

**Comment by GarryFlemings** [comment]
Sorry for the slow response.

I looked through `mj-carousel` HTML code for other such labels and found none I considered possibly relevant for styling. I've never heard of any other reason to have an interest in using classes like this.

I have never felt the need to style other MJML components. Perhaps others similarly contain classes. I don't know.

If I come across the information, I'll report it.

**Comment by timo-reymann** [comment]
Will be generally available with 1.2.0
